src/content/1.7/contribute/how_to_core.md

```diff
@@ -0,0 +1,119 @@
+
+
+```
+git clone https://github.com/preston/PrestaShop.git
+```
+
+> Of course, you need to replace "preston" by your own nickname here.
+
+Then you can start the effective installation:
+
+```
+cd PrestaShop
+docker-compose up
+```
+
+> The installation can take between 10-15 minutes, don't close the terminal!
+
+You'll see a lot of information displayed in your terminal, you should spot the following ones:
+
+```
+prestashop-git | \n* Installing PrestaShop, this may take a while ...
+prestashop-git | -- Installation successful! --
+prestashop-git | \n* Almost ! Starting web server now\n
+```
+
+At this point, your PrestaShop installation is ready and the website is available at http://localhost:8001.
+
+> You can now close the terminal if you want.
+
+## Make your first contribution
+
+The very first step to create a pull request is to create your own git branch.
+
+Let's say you want to suggest a new feature, like emoticon support everywhere. A correct name for your git branch could be "add-emoticons-support":
+
+```
+git checkout -b "add-emoticons-support"
+```
+
+Then you can start to do changes on PrestaShop Core, and create commits: YaY!
+
+> A good practice is to have meaningful commits labels: it's better to have "corrected type hinting in FooBar" than "fixed stuff". 
+
+### Launch the test suite
+
+Your changes now sounds ok, and you're almost ready to share your changes with the community.
+Before all, you may ensure your changes doesn't break everything: this is why we have a test suite you can use.
+
+You can execute it in your dockerized PrestaShop application without alter your website (it uses a specific database).
+
+```
+docker exec prestashop-git sh tests/check_phpunit.sh
+```
+
+### Publish your contribution on GitHub
+
+Once your changes sounds good and tests pass in your local computer, it's time to publish your work online and make a pull request.
+
+The last thing you need to do in the terminal is to publish your branch on GitHub:
+
+```
+git push origin add-emoticons-support
+```
+
+> You'll need to use your GitHub credentials, this is totally ok.
+
+Then, you can create your Pull Request on GitHub! If you don't know how to do it, you can read [GitHub documentation](https://help.github.com/articles/creating-a-pull-request/).
+
+> Don't forget to complete the contribution table, this is really important for the Core Team to really understand what is the value of your contribution.
+
+
+## Syncing your fork
+
+PrestaShop Core is a really active project with more than 30 contributions accepted per week, so your copy will become outdated
+really fast. To make your own copy up to date with the original project, only few commands are required:
+
+> You need to execute these commands at the root of your copy/fork.
+
+```
+git remote add ps https://github.com/PrestaShop/PrestaShop.git
+git fetch ps
+git rebase -i ps/develop
+git push -f origin develop
```

Comment:
> # Developing directly on develop is a bad practice and we must discourage them to do that so rebase is useless

You should read the entire guide carefully, I've never encouraged to use the develop branch ;)

Reply:
In case, if you don't write anything on your develop branch (it's just a branch to help to start a new branch) you don't have any changes, so you will never create a new commit :) 
The rebase is only if you create a commit before, or merge something locally.

And yes, could be a good idea to say clearly "Watch out guys, don't use develop, 1.7.x, etc"

Reply:
It's because I think the contributor will create his feature branch everytime from develop or 1.7.x ... this is what I do, so I rebase develop on a daily task, sometimes twice a day ^^

Reply:
Huhu, bad guy! Maybe one of my favorite way to contribute https://danielkummer.github.io/git-flow-cheatsheet/

Reply:
lol ! You follow exactly the same workflow except you don't know what exactly happens behind "flow" commands ahaha :trollface: 

Troll apart, I'm not in favor of adding "git flow" as requirement even if it encourages good practices :)

Reply:
Haha I always know what I write behind :D (and git flow not use git rebase when merging :roll_eyes: )
Yeah, git flow is better for agency or internal use, not for community of PHP developpers.